int32 tag_id
float32[] points
